library/std/src/sys/sync/rwlock/queue.rs

```diff
@@ -173,7 +175,7 @@ impl AtomicLink {
     }
 }
 
-#[repr(align(8))]
+#[repr(align(16))]
```

Comment:
Could you add a note about alignment? Just that 16 is needed to ensure the last 4 bits of the pointer are always null.

Reply:
I added:

```rust
/// Note that we need an alignment of 16 to ensure that the last 4 bits of any
/// pointers to `Node`s are always zeroed (for the bit flags described in the
/// module-level documentation).
```